airflow/providers/cncf/kubernetes/hooks/kubernetes.py

```diff
@@ -127,6 +127,9 @@ def __init__(
         self.disable_verify_ssl = disable_verify_ssl
         self.disable_tcp_keepalive = disable_tcp_keepalive
 
+        # Expose whether the hook is configured to use incluster_config or not
+        self.is_in_cluster: Optional[bool] = None
```

Comment:
So... maybe what you should do is this

```
@property
def is_in_cluster(self):
    self.client
    return self._is_in_cluster
```

that way it will always tell the truth....

sorta like we arrived at with KPO

Reply:
That's a good idea, thanks 🍺